pkg/backup/backup_manager.go

```diff
@@ -76,54 +73,33 @@ func (bm *BackupManager) SaveSnap(s3Path string) error {
 	return nil
 }
 
-// etcdClientWithMaxRevision gets the etcd member with the maximum kv store revision
+// etcdClientWithMaxRevision gets the etcd endpoint with the maximum kv store revision
 // and returns the etcd client and the rev of that member.
 func (bm *BackupManager) etcdClientWithMaxRevision() (*clientv3.Client, int64, error) {
-	podList, err := bm.kubecli.Core().Pods(bm.namespace).List(k8sutil.ClusterListOpt(bm.clusterName))
-	if err != nil {
-		return nil, 0, err
-	}
-
-	var pods []*v1.Pod
-	for i := range podList.Items {
-		pod := &podList.Items[i]
-		if pod.Status.Phase == v1.PodRunning {
-			pods = append(pods, pod)
-		}
+	client, rev := getEndpointWithMaxRev(bm.endpoints, bm.etcdTLSConfig)
```

Comment:
We probably don't need such complexity of sorting.
Just pass the endpoints as the same as `--endpoints` param?

Reply:
so let etcd client decides then. that makes sense.

Reply:
Cool. 👍

Reply:
why dont we need this logic? how will etcd client library help us to get the same thing done?

Reply:
I guess the main concern here is should etcd back operator be smart on what endpoint to choose or not. let's figure that out first @hongchaodeng  @xiang90 .

Reply:
it seems like we are going to keep the maxRev logic. i'll do that then.